src/app/session/session.service.ts

```diff
@@ -75,42 +75,42 @@ export class SessionService {
       databaseURL: environment.FIREBASE_DATABASE_URL,
       projectId: environment.FIREBASE_PROJECT_ID,
       storageBucket: environment.FIREBASE_STORAGE_BUCKET,
-      messagingSenderId: environment.FIREBASE_SENDER_ID
+      messagingSenderId: environment.FIREBASE_SENDER_ID,
     });
   }
 
-  expired(): boolean {
+  public expired(): boolean {
     return !localStorage.getItem(REFRESH_TOKEN) || this.expires() <= new Date();
   }
 
-  expires(): Date {
+  public expires(): Date {
     return new Date(JSON.parse(localStorage.getItem(REFRESH_TOKEN)).exp * 1000);
   }
 
-  authorizationExpired(): boolean {
+  public authorizationExpired(): boolean {
     return !localStorage.getItem(AUTHORIZATION_TOKEN) || this.authorizationExpires() <= new Date();
   }
 
-  authorizationExpires(): Date {
+  public authorizationExpires(): Date {
     const payload = JSON.parse(atob(localStorage.getItem(AUTHORIZATION_TOKEN).split('.')[1]));
     return new Date(payload.exp * 1000);
   }
 
 
-  refresh() {
+  public refresh(): Subscription {
     const refreshToken = JSON.parse(localStorage.getItem(REFRESH_TOKEN)).val;
     return this.http.post(`${environment.IAM_API}/refresh`, `refresh_token=${refreshToken}`, {
       headers: {'Content-Type': 'application/x-www-form-urlencoded'},
     }).subscribe((response) => {
-      localStorage.setItem(AUTHORIZATION_TOKEN, (<any>response).data);
```

Comment:
Let's set the response type to something more meaningful. I think `Response` from `'@angular/core/http'` would work